FreeTubeApp/FreeTube
# Add setting to auto load comment section

## Pull Request Type


## Related issue



N/A

## Description

Add a setting to allow comment section to be auto loaded, default disabled
When enabled (with a percentage), comment section **bottom** when appears pass N% above bottom of viewport will have comment data auto loaded **once** (won't load it if data already being loaded, loaded)

## Screenshots 

New setting position (can be adjusted)
![image](https://user-images.githubusercontent.com/1018543/227416555-455c14ed-0d55-4946-95f7-9d8121c45ef0.png)


## Testing 



A. Disabled
- No setting change should be required, it's default
- View a video with comment
- Scroll~~~
- Ensure comment not auto loaded

B. 0% from window/view port bottom
- Update setting
- View a video with comment
- Scroll carefully and observe comment section
- Ensure comment auto loaded once comment section **bottom** is 0% above bottom of window/view port

C. 50% from window/view port bottom
- Update setting
- View a video with comment
- Scroll carefully and observe comment section
- Ensure comment auto loaded once comment section **bottom** is 50% above bottom of window/view port

You can test other settings as well, those percentage values (yes zero or negative values) are just passed into [`IntersectionObserver`](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver) [`rootMargin`](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver/rootMargin)

## Desktop

- **OS:**
- **OS Version:**
- **FreeTube version:**

## Additional context

- To prevent comment autoload on page load (when video player is minimized for a moment), a timeout is used (1000ms)
  I am not sure if that's good enough
- Updating setting won't adjust behaviour until video switched
